lib/jwe.c

```diff
@@ -357,6 +359,11 @@ jose_jwe_dec(jose_cfg_t *cfg, const json_t *jwe, const json_t *rcp,
 {
     json_auto_t *cek = NULL;
 
+    if (ptl && *ptl > MAX_COMPRESSED_SIZE) {
```

Comment:
Not sure this is the right place, as the other APIs are not calling into this one and have no checks.

Ideally this check is done once you determined that the payload is actually compressed, so as close as possible to the call of the deflate functions.

You certainly do not want to prevent checking a non-compressed payload that is bigger than what you accept for compressed ones, because there is no DoS amplification to care for in that case.

Reply:
Hello @simo5 . Thanks for detecting this. You are completely right. I will re code it.